packages/core/src/sso/SamlConnector/index.ts

```diff
@@ -0,0 +1,174 @@
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  type GetSession,
+  type SetSession,
+} from '@logto/connector-kit';
+import { assert, appendPath } from '@silverhand/essentials';
+import * as saml from 'samlify';
+import { z } from 'zod';
+
+import { EnvSet, getTenantEndpoint } from '#src/env-set/index.js';
+
+import {
+  type BaseSamlConfig,
+  type BaseSamlConnectorConfig,
+  attributeMappingPostProcessor,
+} from '../types/saml.js';
+
+import {
+  fetchSamlConfig,
+  getRawSamlConfig,
+  getUserInfoFromRawUserProfile,
+  samlAssertionHandler,
+} from './utils.js';
+
+class SamlConnector {
+  private readonly _acsUrl: string;
+  constructor(
+    private readonly config: BaseSamlConnectorConfig,
+    tenantId: string,
+    ssoConnectorId: string
+  ) {
+    this._acsUrl = appendPath(
+      getTenantEndpoint(tenantId, EnvSet.values),
+      `api/authn/saml/sso/${ssoConnectorId}`
+    ).toString();
+  }
+
+  get acsUrl() {
+    return this._acsUrl;
+  }
+
+  /* Fetch SAML config from the metadata XML file or metadata URL. Throws error if config is invalid. */
+  getSamlConfig = async (): Promise<BaseSamlConfig> => {
+    const samlConfig = await fetchSamlConfig(this.config);
+
+    return {
+      ...samlConfig,
+      ...this.config,
+    };
+  };
+
+  getIdpMetadata = async () => {
+    return getRawSamlConfig(this.config);
+  };
+
+  getAuthorizationUrl = async (
```

Comment:
Unlike OIDC, `getAuthorizationUrl` and `getUserInfo` is not SAML standard flow. So these two method should be defined in the SamlSsoConnector.  The basic SamlConnector could have methods that following SAML conventions, like getSingleSignOnUrl, and get parseSamleAssertion.

Further thinking `getSingleSignOnUrl` might be more general for the SsoConnectors. Make be we should update the abstract class SingleSignOn to have public method `getSingleSignOnUrl`.  Add a method mapping for the OIDCConnector instead. 

```
getSingleSignOnUrl(attr) {
 return this.getAuthorizationUrl(attr);
}
```